src/HasTags.php

```diff
@@ -55,16 +55,20 @@ public function setTagsAttribute($tags)
 
     /**
      * @param \Illuminate\Database\Eloquent\Builder $query
-     * @param array|\ArrayAccess|\Spatie\Tags\Tags $tags
+     * @param array|\ArrayAccess|\Spatie\Tags\Tag   $tags
+     * @param string|null                           $type
+     * @param bool                                  $not
      *
      * @return \Illuminate\Database\Eloquent\Builder
      */
-    public function scopeWithAllTags(Builder $query, $tags, string $type = null): Builder
+    public function scopeWithAllTags(Builder $query, $tags, string $type = null, bool $not = false): Builder
```

Comment:
We tend to avoid passing booleans to methods because in makes it a bit unreadable when viewing this from the caller's side.

Remove the boolean and introduce dedictated `not` methods instead.

Reply:
@freekmurze Will do. I just mirrored Laravel's handling of whereIn and whereNotIn, etc. You'd rather have to slightly un-DRY code then?

Reply:
@freekmurze Ping! — the `withoutAllTags` method *is* the "not" method; splitting them up will result in a nearly identical method. Is that what you'd prefer?

Reply:
> You'd rather have to slightly un-DRY code then?

Yeah, that's no problem. I prefer readable method names over being 100% dry.

Reply:
@HipsterJazzbo @freekmurze Any update on this? :-) I'm actually looking for this exact functionality :-)

Reply:
I'm waiting on this PR being updated.